<?xml version="1.0"?>

<launch>

<node pkg="tf" type="static_transform_publisher" name="world_pubsdfsdf" args="0 0 0 0 0 0 global map 30"/>
 <!-- <node pkg="tf" type="static_transform_publisher" name="world_pub" args="0 0 0 0 0 0 global world 30"/> -->


  <node pkg="tf" type="static_transform_publisher" name="map_pub" args="0 0 0 0 0 0 map flat_layer_map1_frame 30"/>

  <arg name="map_file1" default="$(find arips_navigation)/maps/appartement_lu.yaml"/>
  <node name="map_server1" pkg="map_server" type="map_server" args="$(arg map_file1)" >
    <param name="frame_id" type="str" value="flat_layer_map1_frame" />
    <remap from="map" to="flat_layer_map1"/>
    <remap from="map_metadata" to="flat_layer_map1_metadata"/>
  </node>
   
  <node pkg="arips_navigation" type="arips_navigation_node" respawn="false" name="arips_navigation_node" output="screen" clear_params="true">
    <rosparam file="$(find arips_navigation)/config/lu/costmap.yaml" command="load" />
    <rosparam file="$(find arips_navigation)/config/config.yaml" command="load" />
    <rosparam param="topo_planner/topo_map_file" subst_value="True">$(find arips_navigation)/config/lu/topomap.yaml</rosparam>
  </node>



   <!-- Run the map server -->
    <arg name="map_file" default="$(find arips_launch)/../maps/appartement_lu.yaml"/>
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

    <arg name="use_map_topic"   default="true"/>
    <arg name="scan_topic"      default="scan"/>
    <arg name="initial_pose_x"  default="0.0"/>
    <arg name="initial_pose_y"  default="0.0"/>
    <arg name="initial_pose_a"  default="0.0"/>
    <arg name="odom_frame_id"   default="odom"/>
    <arg name="base_frame_id"   default="arips_base"/>
    <arg name="global_frame_id" default="map"/>

    <!--- Run AMCL -->
    <node pkg="amcl" type="amcl" name="amcl">
    	<param name="use_map_topic"             value="$(arg use_map_topic)"/>
      <!-- Publish scans from best pose at a max of 10 Hz -->
      <param name="odom_model_type"           value="diff"/>
      <param name="odom_alpha5"               value="0.1"/>
      <param name="gui_publish_rate"          value="10.0"/>
      <param name="laser_max_beams"             value="60"/>
      <param name="laser_max_range"           value="12.0"/>
      <param name="min_particles"             value="500"/>
      <param name="max_particles"             value="2000"/>
      <param name="kld_err"                   value="0.05"/>
      <param name="kld_z"                     value="0.99"/>
      <param name="odom_alpha1"               value="0.2"/>
      <param name="odom_alpha2"               value="0.2"/>
      <!-- translation std dev, m -->
      <param name="odom_alpha3"               value="0.2"/>
      <param name="odom_alpha4"               value="0.2"/>
      <param name="laser_z_hit"               value="0.5"/>
      <param name="laser_z_short"             value="0.05"/>
      <param name="laser_z_max"               value="0.05"/>
      <param name="laser_z_rand"              value="0.5"/>
      <param name="laser_sigma_hit"           value="0.2"/>
      <param name="laser_lambda_short"        value="0.1"/>
      <param name="laser_model_type"          value="likelihood_field"/>
      <!-- <param name="laser_model_type" value="beam"/> -->
      <param name="laser_likelihood_max_dist" value="2.0"/>
      <param name="update_min_d"              value="0.25"/>
      <param name="update_min_a"              value="0.2"/>
      <param name="odom_frame_id"             value="$(arg odom_frame_id)"/>
      <param name="base_frame_id"             value="$(arg base_frame_id)"/>
      <param name="global_frame_id"           value="$(arg global_frame_id)"/>
      <param name="resample_interval"         value="1"/>
      <!-- Increase tolerance because the computer can get quite busy -->
      <param name="transform_tolerance"       value="1.0"/>
      <param name="recovery_alpha_slow"       value="0.0"/>
      <param name="recovery_alpha_fast"       value="0.0"/>
      <param name="initial_pose_x"            value="$(arg initial_pose_x)"/>
      <param name="initial_pose_y"            value="$(arg initial_pose_y)"/>
      <param name="initial_pose_a"            value="$(arg initial_pose_a)"/>
      <remap from="scan"                      to="$(arg scan_topic)"/>
    </node>

</launch>
